### ISSUE TITLE
certificate renewal fails with urn:acme:error:malformed

### ISSUE THREAD

**Issue by bjmgeek** [lukas2511/dehydrated#428]
I have dehydrated running in a cron script, so I haven't changed my config in a while.  However, it recently stopped working.  I ran it manually, and got the following error:

````
  + ERROR: An error occurred while sending post-request to https://acme-v01.api.letsencrypt.org/acme/challenge/LxPiZ1RIVyEMv-l964GcUA9PU71kbXGPvxQ1p7Vo41g/1894311304 (Status 400)

Details:
{
  "type": "urn:acme:error:malformed",
  "detail": "Unable to update challenge :: The challenge is not pending.",
  "status": 400
}

````

**Comment by lukas2511** [comment]
I don't think I have ever seen that. Have you tried updating dehydrated?

**Comment by bjmgeek** [comment]
Yes, this was from the git master.

**Comment by ludeeus** [comment]
I'm seeing this to :/
Any soulution to it?

**Comment by cpu** [comment]
@ludeeus Can you provide the full output you're seeing? Ideally with the authorization/challenge URL?

The example from @bjmgeek is too old for me to find in the server-side logs for the Let's Encrypt validation authority.

If I had to guess, I'd say this might be related to [a recent (Aug 31st) change to recycle pending authorizations](https://community.letsencrypt.org/t/automatic-recycling-of-pending-authorizations/41321).

@lukas2511 Is there any chance that two threads/instances of Dehydrated could be requesting an authorization, get back the same one, and then trip over each other when one instance POSTs the authorization changing the state from pending->[invalid/valid] before the second instance POSTs the same authorization, receiving the error about it being in a non-pending state?

**Comment by lukas2511** [comment]
@cpu i don't think this is a threading issue, dehydrated should prevent this and in case it doesn't the variance in latencies is so big that probably around the third run one of the instances would just "win" the race condition

@bjmgeek do you by any chance remember if the issue just went away after around 7 days? that would indicate a problem with the recycled authorizations

maybe for some reason dehydrated doesn't notice the verification is already valid and thinks it's still pending, i'll have a look at how that might be possible...

**Comment by cpu** [comment]
> maybe for some reason dehydrated doesn't notice the verification is already valid and thinks it's still pending, i'll have a look at how that might be possible...

That seems like a more plausible explanation than my threading guesses :-)

**Comment by ludeeus** [comment]
@cpu Tried again now, and it worked...
But it looked the same as @bjmgeek

**Comment by bjmgeek** [comment]
Well, let me try again...

**Comment by bjmgeek** [comment]
I'm still seeing an error, but not the malformed one:

````
bminton:~# /etc/cron.weekly/letsencrypt
# INFO: Using main config file /var/local/dehydrated/rsa/config
Processing brian.minton.name with alternative names: www.brian.minton.name secure.brian.minton.name blog.brian.minton.name photos.brian.minton.name music.brian.minton.name bjmgeek.science meadowfen.farm meadowfenfarm.com www.meadowfen.farm www.meadowfenfarm.com meadowfen.farm blog.meadowfen.farm gallery.brian.minton.name bminton.is-a-geek.net caelonna.is-a-geek.net www.bminton.is-a-geek.net www.caelonna.is-a-geek.net
 + Checking domain name(s) of existing cert... changed!
 + Domain name(s) are not matching!
 + Names in old certificate: bjmgeek.science blog.brian.minton.name blog.meadowfen.farm bminton.is-a-geek.net brian.minton.name caelonna.is-a-geek.net gallery.brian.minton.name meadowfen.farm meadowfenfarm.com music.brian.minton.name photos.brian.minton.name secure.brian.minton.name www.bjmgeek.science www.bminton.is-a-geek.net www.brian.minton.name www.caelonna.is-a-geek.net www.meadowfen.farm www.meadowfenfarm.com
 + Configured names: bjmgeek.science blog.brian.minton.name blog.meadowfen.farm bminton.is-a-geek.net brian.minton.name caelonna.is-a-geek.net gallery.brian.minton.name meadowfen.farm meadowfenfarm.com music.brian.minton.name photos.brian.minton.name secure.brian.minton.name www.bminton.is-a-geek.net www.brian.minton.name www.caelonna.is-a-geek.net www.meadowfen.farm www.meadowfenfarm.com
 + Forcing renew.
 + Checking expire date of existing cert...
 + Valid till Aug 24 13:16:00 2017 GMT Certificate will expire
(Less than 30 days). Renewing!
 + Signing domains...
 + Generating private key...
 + Generating signing request...
 + Requesting challenge for brian.minton.name...
 + Already validated!
 + Requesting challenge for www.brian.minton.name...
 + Already validated!
 + Requesting challenge for secure.brian.minton.name...
 + Already validated!
 + Requesting challenge for blog.brian.minton.name...
 + Already validated!
 + Requesting challenge for photos.brian.minton.name...
 + Already validated!
 + Requesting challenge for music.brian.minton.name...
 + Already validated!
 + Requesting challenge for bjmgeek.science...
 + Already validated!
 + Requesting challenge for meadowfen.farm...
 + Requesting challenge for meadowfenfarm.com...
 + Requesting challenge for www.meadowfen.farm...
 + Requesting challenge for www.meadowfenfarm.com...
 + Requesting challenge for meadowfen.farm...
 + Requesting challenge for blog.meadowfen.farm...
 + Requesting challenge for gallery.brian.minton.name...
 + Requesting challenge for bminton.is-a-geek.net...
 + Already validated!
 + Requesting challenge for caelonna.is-a-geek.net...
 + Requesting challenge for www.bminton.is-a-geek.net...
 + Already validated!
 + Requesting challenge for www.caelonna.is-a-geek.net...
 + Responding to challenge for meadowfen.farm...
ERROR: Challenge is invalid! (returned: invalid) (result: {
  "type": "http-01",
  "status": "invalid",
  "error": {
    "type": "urn:acme:error:tls",
    "detail": "Fetching https://meadowfen.farm/.well-known/acme-challenge/YnBrrDhEH50MS_VUtzLWebjXL5Vu3-hmXNVSXFidXik: remote error: tls: handshake failure",
    "status": 400
  },
  "uri": "https://acme-v01.api.letsencrypt.org/acme/challenge/ab7eJAjutzyUyw31yKMkZoRj2rxveUoGhnRPE850pfo/2419593692",
  "token": "YnBrrDhEH50MS_VUtzLWebjXL5Vu3-hmXNVSXFidXik",
  "keyAuthorization": "YnBrrDhEH50MS_VUtzLWebjXL5Vu3-hmXNVSXFidXik.7cmLjPBGSVyPZAIrbFOMeUhOUSmH3n654DW9urqTP6k",
  "validationRecord": [
    {
      "url": "https://meadowfen.farm/.well-known/acme-challenge/YnBrrDhEH50MS_VUtzLWebjXL5Vu3-hmXNVSXFidXik",
      "hostname": "meadowfen.farm",
      "port": "443",
      "addressesResolved": [
        "174.59.12.36"
      ],
      "addressUsed": "174.59.12.36",
      "addressesTried": []
    },
    {
      "url": "http://meadowfen.farm/.well-known/acme-challenge/YnBrrDhEH50MS_VUtzLWebjXL5Vu3-hmXNVSXFidXik",
      "hostname": "meadowfen.farm",
      "port": "80",
      "addressesResolved": [
        "174.59.12.36"
      ],
      "addressUsed": "174.59.12.36",
      "addressesTried": []
    }
  ]
})
# INFO: Using main config file /var/local/dehydrated/ecc/config
Processing brian.minton.name with alternative names: www.brian.minton.name secure.brian.minton.name blog.brian.minton.name photos.brian.minton.name music.brian.minton.name bjmgeek.science meadowfen.farm meadowfenfarm.com www.meadowfen.farm www.meadowfenfarm.com meadowfen.farm blog.meadowfen.farm gallery.brian.minton.name bminton.is-a-geek.net caelonna.is-a-geek.net www.bminton.is-a-geek.net www.caelonna.is-a-geek.net
 + Checking domain name(s) of existing cert... changed!
 + Domain name(s) are not matching!
 + Names in old certificate: bjmgeek.science blog.brian.minton.name blog.meadowfen.farm bminton.is-a-geek.net brian.minton.name caelonna.is-a-geek.net gallery.brian.minton.name meadowfen.farm meadowfenfarm.com music.brian.minton.name photos.brian.minton.name secure.brian.minton.name www.bjmgeek.science www.bminton.is-a-geek.net www.brian.minton.name www.caelonna.is-a-geek.net www.meadowfen.farm www.meadowfenfarm.com
 + Configured names: bjmgeek.science blog.brian.minton.name blog.meadowfen.farm bminton.is-a-geek.net brian.minton.name caelonna.is-a-geek.net gallery.brian.minton.name meadowfen.farm meadowfenfarm.com music.brian.minton.name photos.brian.minton.name secure.brian.minton.name www.bminton.is-a-geek.net www.brian.minton.name www.caelonna.is-a-geek.net www.meadowfen.farm www.meadowfenfarm.com
 + Forcing renew.
 + Checking expire date of existing cert...
 + Valid till Aug 24 13:17:00 2017 GMT Certificate will expire
(Less than 30 days). Renewing!
 + Signing domains...
 + Generating private key...
 + Generating signing request...
 + Requesting challenge for brian.minton.name...
 + Already validated!
 + Requesting challenge for www.brian.minton.name...
 + Already validated!
 + Requesting challenge for secure.brian.minton.name...
 + Already validated!
 + Requesting challenge for blog.brian.minton.name...
 + Already validated!
 + Requesting challenge for photos.brian.minton.name...
 + Already validated!
 + Requesting challenge for music.brian.minton.name...
 + Already validated!
 + Requesting challenge for bjmgeek.science...
 + Already validated!
 + Requesting challenge for meadowfen.farm...
 + Requesting challenge for meadowfenfarm.com...
 + Requesting challenge for www.meadowfen.farm...
 + Requesting challenge for www.meadowfenfarm.com...
 + Requesting challenge for meadowfen.farm...
 + Requesting challenge for blog.meadowfen.farm...
 + Requesting challenge for gallery.brian.minton.name...
 + Requesting challenge for bminton.is-a-geek.net...
 + Already validated!
 + Requesting challenge for caelonna.is-a-geek.net...
 + Requesting challenge for www.bminton.is-a-geek.net...
 + Already validated!
 + Requesting challenge for www.caelonna.is-a-geek.net...

 + Responding to challenge for meadowfen.farm...
ERROR: Challenge is invalid! (returned: invalid) (result: {
  "type": "http-01",
  "status": "invalid",
  "error": {
    "type": "urn:acme:error:tls",
    "detail": "Fetching https://meadowfen.farm/.well-known/acme-challenge/xFiN_DW0fpXNYvMkEmGFUJOd4qjDPu-OPImav527voA: remote error: tls: handshake failure",
    "status": 400
  },
  "uri": "https://acme-v01.api.letsencrypt.org/acme/challenge/4WYYga7hQzhVu76olce4VHCTTFLPDO6p5pwe0xlxZ7o/2419595459",
  "token": "xFiN_DW0fpXNYvMkEmGFUJOd4qjDPu-OPImav527voA",
  "keyAuthorization": "xFiN_DW0fpXNYvMkEmGFUJOd4qjDPu-OPImav527voA.-EpzKSq6tAP-GpNV-kX5q4Wez8B8Ru1P8-XZkpIcgN8",
  "validationRecord": [
    {
      "url": "https://meadowfen.farm/.well-known/acme-challenge/xFiN_DW0fpXNYvMkEmGFUJOd4qjDPu-OPImav527voA",
      "hostname": "meadowfen.farm",
      "port": "443",
      "addressesResolved": [
        "174.59.12.36"
      ],
      "addressUsed": "174.59.12.36",
      "addressesTried": []
    },
    {
      "url": "http://meadowfen.farm/.well-known/acme-challenge/xFiN_DW0fpXNYvMkEmGFUJOd4qjDPu-OPImav527voA",
      "hostname": "meadowfen.farm",
      "port": "80",
      "addressesResolved": [
        "174.59.12.36"
      ],
      "addressUsed": "174.59.12.36",
      "addressesTried": []
    }
  ]
})
bminton:~# 
````

**Comment by lukas2511** [comment]
@bjmgeek this looks like a completely unrelated issue, which seems to be on your side. maybe it's a problem with the expired certificate and your forced ssl redirect

**Comment by bjmgeek** [comment]
Perhaps.  In any case, I'm trying with a dns-01 challenge now.

**Comment by lukas2511** [comment]
Since there were no more reports of this happening (and the relevant code has been restructured anyway) I consider this issue solved.
